### DISCUSSION TITLE
Duplicate / Clone Plan to save copy-paste

### DISCUSSION THREAD

**Discussion by semiligneous** [garethgeorge/backrest#425] [Ideas]
It would be nice if there was a way to duplicate a plan, make edits and then save it as a new plan. I want to set up a plan per docker container that I want backed up, and the majority of settings will be identical each time, or at least very similar.  Including custom commands as hooks pre- and post-backup. 

Lots of copying and pasting each time from a 2nd tab could be saved if there was a button inside the "Update Plan" modal which said "copy plan" (perhaps right at the bottom next to "Submit"?), and it opened a new modal, with the Plan Name as "old plan (copy)" and all the other details identical - cron/schedule/path/retention policy/hooks etc etc.

That way, it would be much quicker and less error-prone to set up multiple backup plans, and when a new docker container is added, it would be 3 or 4 clicks to set up the backup in a consistent way.

Is there another way to achieve this?

**Comment by semiligneous** [comment]
I can answer myself, one "other way" would be to manually edit the .config/backrest/config.json and repeat the relevant plan blocks.

That works fine. (seems to need to restart the systemd service to pick up the config change though)

Not super hard, but open to breaking things, and given this is a GUI tool, it seems less than ideal to go back to manual json editing! But it works.

A UI would be nicer, not sure if anyone else agrees or shares the need?

**Comment by clnhlzmn** [comment]
I just discovered this deficiency and I fully agree. I'd even settle for the "Plan config as json" view being editable so I could at least just copy the json from one plan, create a new plan, paste the json, and then graphically tweak the things I want to tweak.

**Reply by lzecca78** [reply]
Agree with this, it also would be amazing for respecting the 3-2-1 backup strategy. What I use for example the copy paste is to address the same backup to different destinations (local and cloud) when should be amazing having the feature to assign the same backup to different destinations, which could be way easier.

**Reply by hermannx5** [reply]
I also agree, that would be amazing...
